src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -776,7 +780,12 @@ public async Task VersionHandshakeAsync(NetworkPeerRequirement requirements, Can
 
                             this.logger.LogDebug("Sending version acknowledgement.");
                             await this.SendMessageAsync(new VerAckPayload(), cancellationToken).ConfigureAwait(false);
-                            this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(versionPayload.AddressFrom, false);
+
+                            // Note that we only update our external address data from information returned by outbound peers.
+                            // TODO: Is this due to a security assumption or is it an oversight? There is a higher risk the inbounds could be spoofing what they claim our external IP is. We would then use it in future version payloads, so that could be considered an attack.
+                            // For outbounds: AddressFrom is our current external endpoint from our perspective, and could easily be incorrect if it has been automatically detected from local NICs.
+                            // Whereas AddressReceiver is the endpoint from the peer's perspective, so we update our view using that.
+                            this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(versionPayload.AddressReceiver, false);
```

Comment:
Good spot!

Reply:
Oh so we only set false here cause we are worried that the other node could be setting some other IP in the version payload back to us? Once it is set here you seem to imply that we send this to other nodes going forward?

Reply:
I'm wondering what the potential harm is to set it to true (final)?

Reply:
As discussed, finality is 'special' and reserved for the case where `-externalip` has actually been supplied via config. In normal (non-config) usage it will be determined from the aggregate view of the peers via the score, it never becomes 'final'